### PR TITLE
Signup

### LINKED ISSUE #32: SignUp
To create a user, he must be able to create one.
A view is created. A create is added to the store authentication.
To do this, certain fields must be validated

## Time tracking
* Estimation: 4h
* Effective: . -h

### PR DESCRIPTION
**Pull request by fgervasi**
Fixes #32

### CODE REVIEW

File: src/views/Login.vue [hunk 1]
```diff
@@ -33,6 +33,9 @@
           </v-form>
         </v-card-text>
         <v-card-actions>
+          <v-btn text x-small link :to="{ name: 'signup' }">
+            You don't have an account? To sign up!
```
Comment (TiMESPLiNTER) [inline src/views/Login.vue:37]:
```suggestion
            You don't have an account? Create one now!
```

File: src/views/SignUp.vue [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+              :error="hasConfirmPwdError"
+              :error-messages="confirmPwdErrorMsgs"
+              @input="resetConfirmPassword()"
+              @click:append="showConfirmPasswor = !showConfirmPasswor"
+            />
+          </v-form>
+        </v-card-text>
+        <v-card-actions>
+          <v-btn text x-small link :to="{ name: 'login' }">
+            I am already registered back to login
```
Comment (TiMESPLiNTER) [inline src/views/SignUp.vue:75]:
```suggestion
            You already have an account? Sign in now!
```

File: src/views/SignUp.vue [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      "This value is too short. It should have 3 characters or more.",
+    usernameMin: ($: string) =>
+      $.length >= 5 ||
+      "This value is too short. It should have 5 characters or more.",
+    min: ($: string) => $.length >= 8 || "Min 8 characters",
+    emailRules: ($: string) => /.+@.+\..+/.test($) || "E-mail must be valid",
+    hasLetterRules: ($: string) =>
+      /[A-Za-z]/.test($) || "Your password must contain at least one letter.",
+    hasNumberRules: ($: string) =>
+      /[0-9]/.test($) || "Your password must contain at least one digit.",
```
Comment (TiMESPLiNTER) [inline src/views/SignUp.vue:126]: Frankly I'm not a big fan of password rules other than min length. It just makes you forgetting your password or filling up your password with easy to guess numbers and letters.

File: src/views/SignUp.vue [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  };
+
+  @Ref("signUpForm") readonly form!: HTMLFormElement;
+
+  @auth.Action
+  public createUser!: (user: CreateUser) => Promise<void>;
+
+  public handleSubmit() {
+    if (this.password !== this.confirmPassword) {
+      this.confirmPwdErrorMsgs.push("Doesn't match Password");
```
Comment (TiMESPLiNTER) [inline src/views/SignUp.vue:136]:
```suggestion
      this.confirmPwdErrorMsgs.push("Passwords don't match");
```

File: src/views/SignUp.vue [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      this.hasConfirmPwdError = true;
+      return;
+    }
+    if (this.form.validate()) {
+      this.createUser({
+        username: this.user,
+        displayName: "asfsa",
```
Comment (TiMESPLiNTER) [inline src/views/SignUp.vue:143]: what's this? 😅 I wouldn't give the use an option to define their displayName upon sign up. A user can change it later by editing their profile. Or is it required for signup?

File: src/views/home/post/Posts.vue [hunk 6]
```diff
@@ -1,44 +1,57 @@
...
+            <v-spacer></v-spacer>
+            <v-btn icon disabled> <v-icon>mdi-comment-outline</v-icon> </v-btn>
+            <v-btn icon disabled>
+              <v-icon>mdi-heart-circle-outline</v-icon>
+            </v-btn>
+          </v-card-actions>
+        </v-card>
+      </div>
+    </div>
+    <div v-else>
```
Comment (TiMESPLiNTER) [inline src/views/home/post/Posts.vue:46]: Empty state 🎉

File: src/views/SignUp.vue (hunk 5 above)
Comment (fgervasi) [reply src/views/SignUp.vue:143]: :see_no_evil:

File: .env [hunk 7]
```diff
@@ -1,3 +1,3 @@
 VUE_APP_API_CLIENT="server"
 BACKEND_URL="http://localhost:1337"
-#BACKEND_URL="https://mitra-social.herokuapp.com"
\ No newline at end of file
+# BACKEND_URL="https://mitra-social.herokuapp.com"
```
Comment (TiMESPLiNTER) [inline .env:3]: don't think we should commit this?

File: .env.local  [hunk 8]
```diff
@@ -0,0 +1,3 @@
+VUE_APP_API_CLIENT="server"
+BACKEND_URL="http://localhost:1337"
+# BACKEND_URL="https://mitra-social.herokuapp.com"
```
Comment (TiMESPLiNTER) [inline .env.local :3]: you really want to commit this?

Comment (fgervasi) [reply .env.local :3]: Strange, Actually the whole file shouldn't be checked in, in the gitignore it is registered for ignoring it anyway

**Pull request merged**